R-package/tests/testthat/test_Predictor.R

```diff
@@ -187,8 +187,14 @@ test_that("Feature contribution predictions do not take non-general CSR or CSC i
       , params = list(min_data_in_leaf = 5L, num_threads = .LGB_MAX_THREADS)
     )
 
-    expect_error(predict(bst, SmatC, type = "contrib"))
-    expect_error(predict(bst, SmatR, type = "contrib"))
+    expect_error(
+      predict(bst, SmatC, type = "contrib"),
+      regexp = "Predictions on sparse inputs are only allowed for 'dsparseVector', 'dgRMatrix', 'dgCMatrix' - got: dsCMatrix"  # nolint: line_length.
+    )
+    expect_error(
+      predict(bst, SmatR, type = "contrib"),
+      regexp = "Predictions on sparse inputs are only allowed for 'dsparseVector', 'dgRMatrix', 'dgCMatrix' - got: dsRMatrix" # nolint: line_length.
+    )
```

Comment:
```suggestion
    expect_error(
      predict(bst, SmatC, type = "contrib")
      , regexp = "Predictions on sparse inputs are only allowed for 'dsparseVector', 'dgRMatrix', 'dgCMatrix' - got: dsCMatrix"  # nolint: line_length
    )
    expect_error(
      predict(bst, SmatR, type = "contrib"),
      , regexp = "Predictions on sparse inputs are only allowed for 'dsparseVector', 'dgRMatrix', 'dgCMatrix' - got: dsRMatrix" # nolint: line_length
    )
```

These `no_lint` comments don't need periods at the end. It might not matter, but I'd like to remove those since we don't use them anywhere else:

https://github.com/microsoft/LightGBM/blob/618cfafab66d19aced9878ac00c27d592a773bb4/R-package/R/utils.R#L6

To protect us from any parsing bugs in future versions of `{lintr}` which might not ignore those.

Also, we use comma-first style for R code in this project. Please keep this code consistent with that.

Reply:
No worries, and that's good to know about mypy!

There were a couple extra argument commas, which I've removed in [de1774f](https://github.com/microsoft/LightGBM/pull/6931/commits/de1774f07243e672c9ffb3b79e5509781a6d3207). And I'd be happy to do a follow-up PR adding those `nolint` periods in, I'll open that once this gets merged.